alf/utils/dist_utils.py

```diff
@@ -1215,14 +1220,14 @@ def get_base_dist(dist):
         NotImplementedError: if ``dist`` or its based distribution is not
             ``td.Normal``, ``td.Independent`` or ``td.TransformedDistribution``.
     """
-    if isinstance(dist, (td.Normal, td.Categorical, StableCauchy, Beta,
-                         TruncatedDistribution)):
+    dist_type = type(dist)
+    if dist_type in _base_dist_types:
         return dist
```

Comment:
This is not right? The base dist of a Diag is not itself.

Reply:
I guess I misunderstood the purpose of this function then? To me the base dist should be the outermost non-transformed distribution. 

Otherwise, by that same logic, if this function is trying to find the lowest-level (in terms of inheritance or attribute) distribution, then why would it stop at `StableCauchy` and not continue onto `td.Cauchy`?

Reply:
base_dist is the innermost wrapped distribution.

Reply:
I've reverted this and changed the purpose of this PR to the arg bug fix and minor fixes.